sdcm/nemesis.py

```diff
@@ -1716,6 +1712,33 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def disrupt_add_del_nodes(self):
+        add_nodes_number = self.tester.params.get('add_node_cnt', default=1)
+        init_nodes_number = self.tester.params.get("n_db_nodes")
+
+        max_nodes_number = add_nodes_number + init_nodes_number
+        current_nodes_number = len(self.cluster.nodes)
+
+        if max_nodes_number - current_nodes_number > 0:
+            self._set_current_disruption("AddNodes")
+            while len(self.cluster.nodes) < max_nodes_number:
+                self.metrics_srv.event_start('add_node')
+                added_node = self._add_and_init_new_cluster_node()
+                added_node.is_seed = False
+                self.metrics_srv.event_stop('add_node')
+                self.log.info("New node added %s", added_node.name)
+            self.log.info("Added %s nodes to cluster", max_nodes_number - current_nodes_number)
+        else:
```

Comment:
This is not exactly what I meant.
My original intention was to grow the cluster to Y, sleep few minutes and reduce it back to X.
In your case, it will happen only if/when this nemesis will be called again.
It could cause our costs to be much higher than we have now, so let's change it.

Reply:
The idea was to run another nemesises on new cluster size. 
Changed to algorithm you suggested.